### ISSUE TITLE
$25: false-positive when nesting TypeScript generic expressions

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1241]
Steps to reproduce:

```ts
import type { useTranslation } from '@tiktok/plugin-runtime/i18n';


export const ContactUsLinks = (t: ReturnType<typeof useTranslation<"translation">>['t']) => {};
```

Expected: should not produce any errors
Actual: errors with `use of undeclared variable: useTranslation [E0057]`

Compare this to https://www.typescriptlang.org/play/?#code/PTCWFsAcHsCcBcAE8CekCmiDeiCuBndAFVgEMA7fAG1PlGnMQF9EAzWacRAcgAE6A1vGgDgkKrgDmocgFpYucnXDowARgAc5bgG4AUABN0AYxqxMxhviQFiZSjToMAXIgA8RROgAe8dOQN8RGtYGUkAPkQACgA6ONJYSXxXIgBtAF0ASgBecKx4V0UBcmgAd3ImfT09HxgEREtKJABhBnhSY3gAVXwAGRkBIOzogsQAJXR4XFhyIjR0N1QMaFY8QhIKalp6cjcAInh7LadyPfDw1O54bizEXOxKvSA
